cluster-autoscaler/simulator/nodes.go

```diff
@@ -75,3 +45,20 @@ func BuildNodeInfoForNode(node *apiv1.Node, podsForNodes map[string][]*apiv1.Pod
 	}
 	return result, nil
 }
+
+func filterRequiredPodsForNode(allPods []*apiv1.Pod) []*apiv1.Pod {
+	var selectedPods []*apiv1.Pod
+
+	for id, pod := range allPods {
+		// Ignore pod in deletion phase
+		if pod.DeletionTimestamp != nil {
+			continue
+		}
+
+		if pod_util.IsMirrorPod(pod) || pod_util.IsDaemonSetPod(pod) || pod_util.IsStaticPod(pod) {
```

Comment:
IIUC the mirror pod is manifestation of static pod registered in API server. Static pod is observed only by kubelet so we should not get a static pod ever here. Right?

https://kubernetes.io/docs/tasks/configure-pod-container/static-pod/

Reply:
absolutely